sympy/core/power.py

```diff
@@ -1703,36 +1708,29 @@ def mul(d1, d2):
         if not (e.is_integer and e.is_positive and (e*d - n).is_nonpositive and
                 res == _mexpand(self)):
             res += O(x**n, x)
-        return _mexpand(res)
+        return res
 
-    def _eval_as_leading_term(self, x, cdir=0):
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
         from ..series import Order
-        from sympy import exp, I, im, log
+        from sympy import exp, I, im, log, PoleError
         e = self.exp
         b = self.base
         if self.base is S.Exp1:
-            arg = self.exp
-            if arg.is_Add:
-                return Mul(*[(S.Exp1**f).as_leading_term(x) for f in arg.args])
-            arg_1 = arg.as_leading_term(x)
-            if Order(x, x).contains(arg_1):
+            arg = e.as_leading_term(x, logx=logx)
+            if Order(x, x).contains(arg):
                 return S.One
-            if Order(1, x).contains(arg_1):
-                return S.Exp1**arg_1
-            ####################################################
-            # The correct result here should be 'None'.        #
-            # Indeed arg in not bounded as x tends to 0.       #
-            # Consequently the series expansion does not admit #
-            # the leading term.                                #
-            # For compatibility reasons, the return value here #
-            # is the original function, i.e. exp(arg),         #
-            # instead of None.                                 #
-            ####################################################
-            return S.Exp1**arg
+            if Order(1, x).contains(arg):
```

Comment:
This could probably be `arg.subs(x, 0).is_finite`. (`arg` should be well-behaved when `as_leading_term` does not raise.)